### PR TITLE
Add tools/compat-dump.el which generates data/compat-symbols

### PR DESCRIPTION
**Pull request by minad**
Load package-lint--compat-symbols from data/compat-symbols. Drop Compat dependency.

### CODE REVIEW

File: package-lint.el [hunk 1]
```diff
@@ -172,33 +172,17 @@ symbol such as `variable-added'.")
...
-                                            (group (+? any))
-                                            symbol-end)
-                                        nil t)
-            (pcase (match-string 1)
-              ("defvar" (push (intern (match-string 2)) symbols))
-              ((or "defun" "defmacro" "defalias") (push (intern (match-string 2)) functions)))))))
-    (cons symbols functions))
+(defconst package-lint--compat-symbols
+  (with-temp-buffer
+    (insert-file-contents (expand-file-name "data/compat-symbols"))
```
Comment (purcell) [inline package-lint.el:177]: This won't always find the file, so you should probably extract or duplicate the code that loads the sym-dump data

**Comment by purcell** [comment]
That was quick :grin

**Comment by minad** [comment]
> That was quick 😁

Well, all I had to do is to copy your extraction code. ;)

File: package-lint.el (hunk 1 above)
Comment (minad) [reply package-lint.el]: Right. I extracted a function `package-lint--load-data` which is used to load both data/compat-symbols and data/stdlib-changes.

**Pull request merged**